order_processor/function_app.py

```diff
@@ -0,0 +1,127 @@
+import azure.functions as func
+import azure.durable_functions as df
+import logging
+from time import sleep 
+from models import *
+
+myApp = df.DFApp(http_auth_level=func.AuthLevel.ANONYMOUS)
+
+@myApp.route(route="orchestrators/{functionName}")
+@myApp.durable_client_input(client_name="client")
+async def http_start(req: func.HttpRequest, client):    
+    # Request an order
+    order_payload = OrderPayload(
+        order_name="milk", 
+        total_cost=5, 
+        quantity=1
+    )  
+    orchestrator_name = req.route_params['functionName']
+    instance_id = await client.start_new(orchestrator_name, client_input=order_payload)
+    
+    logging.info(f"Started orchestration with ID = '{instance_id}'.")
+
+    response = client.create_check_status_response(req, instance_id)
+    return response
+
+@myApp.orchestration_trigger(context_name="context")
+def process_orchestrator(context: df.DurableOrchestrationContext):   
+    # Retry options for activity functions
+    retry_interval_in_milliseconds = 2000
+    max_number_of_attempts = 3
+    retry_options = df.RetryOptions(retry_interval_in_milliseconds, max_number_of_attempts)
+     
+    # Reserve requested inventory 
+    order_id = context.instance_id
+    order_payload: OrderPayload = context.get_input()
+    inventory_request = InventoryRequest(
+        request_id=order_id, 
+        item_name=order_payload.order_name,
+        quantity=order_payload.quantity
+    )
+    
+    inventory_result: InventoryResult = yield context.call_activity_with_retry("reserve_inventory", retry_options, inventory_request)   
+    
+    # There's not enough inventory, so fail and notify the customer
+    if not inventory_result.success:
+        notification = Notification(f"Insufficient inventory for {order_payload.order_name}.")
+        yield context.call_activity("notify_customer", notification)

```

Comment:
Hey.  could we please confirm that we want `yield` operators here for async calls to durable function activities from the orchestration?  That wasn't immediately intuitive to me but I don't fully understand the dynamics here.

Reply:
we discussed and confirmed that `yield` is appropriate here.